engine/mobility_int.h

```diff
@@ -84,7 +89,14 @@ typedef struct mobile_engfuncs_s
 
 	// COM_ParseFile but with buffer size limit, len reports written size or -1 on overflow
 	char* (*pfnParseFile)( char *data, char *buf, const int size, unsigned int flags, int *len );
-	// To be continued...
+
+	// Functions provided for making possible implementing custom GUI on client side
+	int (*pfnGetClipboardText)( char *buffer, int bufferSize );
+	void (*pfnSetClipboardText)( const char *text );
+	void (*pfnSupressActivityVGUI)( int enable );
```

Comment:
Looks like a clutch rather than a complete solution.

Why we need to suppress VGUI at all? Isn't VGUI is implemented on client side? If you don't want VGUI, just don't have it. Alternatively, add vgui_support interface into client.dll and select whether VGUI or custom UI should be active.

IMHO, most of these functions are duplicates of what vgui_support adds. It's better vgui_api to be extended to support such things like clipboard, text input and so on.

Reply:
> Why we need to suppress VGUI at all?
Извиняюсь, что пишу не на английском, но VGUI мешает получать инпут ивенты где-то на верхнем уровне - они даже не доходят до SDL. Только вот эта supress activity функция решила проблему.
> Isn't VGUI is implemented on client side? If you don't want VGUI, just don't have it. 
PrimeXT вообще никак не использует VGUI по дефолту, там даже библиотека не прилинкована.
> Alternatively, add vgui_support interface into client.dll and select whether VGUI or custom UI should be active.
А вот про это хотелось бы узнать подробнее, лучше сделать вот таким образом, если это более правильный путь

Reply:
Взаимодействовать с VGUI не нужно.

Но вот прокинуть ивенты из платформы в vgui_support -- это благо и вам, и нам. Просто можно расширить его интерфейс аналогичным образом, но предоставлять его движку будет client.dll.

Reply:
> но предоставлять его движку будет client.dll.

Такое уже предусмотрено в движке? Или это просто как вариант реализации?

Reply:
Таки да, но он отключён сейчас для платформ где дллки находятся в доступной файловой системе. 

https://github.com/FWGS/xash3d-fwgs/blob/master/engine/client/vgui/vgui_draw.c#L224

Reply:
А как стоит сделать? Включить по дефолту?

Reply:
Я думаю можно будет включать по дефолту. Если есть InitVGUISupportAPI экспорт, значит берётся оный.

Но тогда это работа по стабилизации API саппорта, ломать его будет нельзя.

Reply:
Я вот не очень понял, как я смогу решить свою задачу при помощи vgui_support интерфейса на стороне клиента. Ну вот я проброшу нужные мне функции, но мне же придётся тогда еще остальные функции копипастить себе из vgui_support библиотеки? Или как

Reply:
Тебе нужно влинковать vgui_support статически и если нужна поддержка vgui, то и опенсорсный vgui влинковать.
Разумеется это костыль.